cli/cmd/build/config/config.go

```diff
@@ -15,13 +16,49 @@ type Artifact struct {
 	URL  string `yaml:"url"`
 }
 
+type Extension struct {
+	Name    string            `yaml:"name"`
+	Version string            `yaml:"version"`
+	Specs   map[string]string `yaml:"specs"`
+}
+
 type Config struct {
-	Name        string     `yaml:"name"`
-	Version     string     `yaml:"version"`
-	LLMAnalyzer bool       `yaml:"llmanalyzer"`
-	Authors     []string   `yaml:"authors"`
-	Categories  []string   `yaml:"categories"`
-	Artifacts   []Artifact `yaml:"artifacts"`
+	Source      string      `yaml:"source"`
+	Name        string      `yaml:"name"`
+	Version     string      `yaml:"version"`
+	LLMAnalyzer bool        `yaml:"llmanalyzer"`
+	Authors     []string    `yaml:"authors"`
+	Categories  []string    `yaml:"categories"`
+	Artifacts   []Artifact  `yaml:"artifacts"`
```

Comment:
we should build config artifacts type object to include metadata + digest string

Reply:
digest should be a string we decode on our end into the object

Reply:
let's update this separately as it falls outside the scope of this PR. also we need to define if this `annotations`/`metadata` field will be allowed when we load config from flags